app/controllers/suggestions_controller.rb

```diff
@@ -0,0 +1,54 @@
+# require '/lib/tunes_takeout_wrapper'
+class SuggestionsController < ApplicationController
+  skip_before_action :require_login, only: :index
+
+  def index
+    @suggestions = extract_suggestions(TunesTakeoutWrapper.top(4))
+    #shows top 20 suggestions, ranked by total number of favorites
+    if current_user
+      @spotify_user = RSpotify::User.find(@current_user.uid)
+      @user = current_user
+      @image = @spotify_user.images[0]["url"]
+    end
+    # if current_user
+    #   @favorites = TunesTakeoutWrapper.user_favorites(@user.user_id)
+    # end
+  end
+
+  def favorites #display user favorites
+    @user = current_user
+    @suggestions = TunesTakeoutWrapper.user_favorites(@user.uid)
+    unless @suggestions.suggestions ==[]
+      @suggestions = extract_suggestions(@suggestions) #returns array of pairing id's, FIX CALL
+    end
+  end
```

Comment:
I'm also noticing that this method doesn't end in rendering a view. Maybe this is a method that belongs in a model?